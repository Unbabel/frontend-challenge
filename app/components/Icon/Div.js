import styled from 'styled-components';

export default styled.div`
  color: ${props => `#${props.color}`};
  max-width: ${props => `${props.size}px`};
`;
