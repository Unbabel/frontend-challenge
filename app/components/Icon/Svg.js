import styled from 'styled-components';

export default styled.svg``;
