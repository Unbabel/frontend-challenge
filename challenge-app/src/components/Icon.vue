<template>
  <picture>
    <source type="image/svg+xml" :srcset="url('svg')" />
    <img :src="url('png', '@1x')" :srcset="`${url('png', '@2x')} 2x`" :alt="description" />
  </picture>
</template>

<script setup>
import { toRefs } from "vue";

const props = defineProps({
  icon: {
    type: String,
    required: true
  },
  description: {
    type: String,
    default: ""
  }
});

const { icon, description } = toRefs(props);

const url = (extension, size = "") =>
  new URL(`../assets/icons/${icon.value}${size}.${extension}`, import.meta.url);
</script>

<style scoped></style>
