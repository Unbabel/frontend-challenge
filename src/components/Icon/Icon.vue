<template>
  <img class="qa-icon" v-if="name.length > 0" :src="computedIcon" />
</template>

<script>
export default {
  name: "Icon",
  props: {
    name: {
      type: String,
      default: "",
    },
  },
  computed: {
    computedIcon() {
      return require(`@/assets/images/${this.name}.svg`);
    },
  },
};
</script>
