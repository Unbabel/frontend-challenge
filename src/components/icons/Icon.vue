<template>
    <component :is="componentName" />
</template>

<script lang="ts">
import { defineAsyncComponent, defineComponent } from 'vue'

export default defineComponent({
    props: ['name'],
    components: {
        UploadIcon: defineAsyncComponent(() => import('@/components/icons/UploadIcon.vue')),
        FetchIcon: defineAsyncComponent(() => import('@/components/icons/FetchIcon.vue')),
        CheckmarkIcon: defineAsyncComponent(() => import('@/components/icons/CheckmarkIcon.vue')),
        PersonIcon: defineAsyncComponent(() => import('@/components/icons/PersonIcon.vue')),
        DeleteIcon: defineAsyncComponent(() => import('@/components/icons/DeleteIcon.vue')),
        AddIcon: defineAsyncComponent(() => import('@/components/icons/AddIcon.vue')),
    },
    computed: {
        componentName() {
            return this.name + '-icon'
        }
    }
})
</script>