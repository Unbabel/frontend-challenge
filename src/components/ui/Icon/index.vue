<template>
  <span :class="name">
    <svg :width="`${width}px`" :height="`${height}px`">
      <use :xlink:href="`#${name}`" />
    </svg>
  </span>
</template>

<script>
export default {
  name: "Icon",

  props: {
    width: {
      type: Number,
      default: 32
    },
    height: {
      type: Number,
      default: 32
    },
    name: {
      type: String,
      require: true,
      default: "person"
    },
    color: {
      type: String
    }
  }
};
</script>

<style scoped lang="scss">
span {
  color: $color-grey;

  &.person {
    color: $color-blue;
  }

  &.upload,
  &.fetch-document {
    color: $color-grey-dark;
  }

  &.add-row {
    color: $color-grey-light;
  }

  &:hover {
    color: $color-purple;
  }
}
</style>
