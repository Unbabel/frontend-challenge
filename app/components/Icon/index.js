import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import PropTypes from 'prop-types';

import { COLORS } from '../../theme';

import Div from './Div';
import Svg from './Svg';

function Icon(props) {
  return (
    <Div {...props} theme={useContext(ThemeContext)}>
      <Svg width={`${props.size}px`} height={`${props.size}px`}>
        <use xlinkHref={`#${props.name}`} />
      </Svg>
    </Div>
  );
}

Icon.propTypes = {
  name: PropTypes.oneOf(['add-row', 'avatar', 'fetch-document', 'person', 'upload']).isRequired,
  color: PropTypes.oneOf([COLORS.BLUE, COLORS.GREY_DARK, COLORS.GREY_LIGHT, COLORS.GREY_LIGHTER]),
  size: PropTypes.number,
  viewBox: PropTypes.string,
};

Icon.defaultProps = {
  color: COLORS.GREY_DARK,
  size: 32,
};

export default Icon;
