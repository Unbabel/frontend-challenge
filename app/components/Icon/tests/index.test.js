import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';

import Icon from '../index';
import configureStore from '../../../configureStore';

describe('<Icon />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a svg', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedRouter history={history}>
            <Icon />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
