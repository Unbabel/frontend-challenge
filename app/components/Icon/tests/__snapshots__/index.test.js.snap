// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render a svg 1`] = `
<svg
  class="Svg-sc-1x9ofvp-0 cKvrhD"
/>
`;
